ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt

```diff
@@ -353,14 +353,16 @@ class GoogleMapsFragment : Hilt_GoogleMapsFragment(), Map {
     }
   }
 
-  private fun addTemplateUrlTileOverlay(url: String) {
-    addTileOverlay(TemplateUrlTileProvider(url))
+  private fun addOfflineTileOverlay(url: String, bounds: List<Bounds>) {
+    addTileOverlay(
+      ClippingTileProvider(TemplateUrlTileProvider(url), bounds.map { it.toGoogleMapsObject() })
+    )
   }
 
   override fun addTileOverlay(tileSource: TileSource) =
     when (tileSource.type) {
       MOG_COLLECTION -> addMogCollectionTileOverlay(tileSource.url)
-      TILED_WEB_MAP -> addTemplateUrlTileOverlay(tileSource.url)
+      TILED_WEB_MAP -> addOfflineTileOverlay(tileSource.url, tileSource.clipBounds!!)
```

Comment:
Should `clipBounds` be nullable?

Reply:
In the current impl, no, but replacing nullable w/default value upstream anyway to prevent regressions here.